### PR TITLE
[fix] Allow application name length to go up to the BOLOS limit

### PR DESCRIPTION
**Pull request by lpascal-ledger**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,16 @@ All notable changes to this project will be documented in this file.
...
 
+## [2.0.12](https://github.com/ledgerhq/app-exchange/compare/2.0.11...2.0.12) - 2022-??-??
+
+### Changed
+
+- Application name length is no longer caped to 15B, but synced with BOLOS maximum length
+
+## [2.0.11](https://github.com/ledgerhq/app-exchange/compare/2.0.3...2.0.11) - 2022-05-24
+
+TODO
```
Comment (fbeutin-ledger) [inline CHANGELOG.md:16]: To do !

File: src/checks.h [hunk 2]
```diff
@@ -0,0 +1,31 @@
+#ifndef _CHECKS_H_
+#define _CHECKS_H_
```
Comment (fbeutin-ledger) [inline src/checks.h:2]: `#pragma once`

File: test/python/apps/exchange.py [hunk 3]
```diff
@@ -186,12 +187,13 @@ def check_transaction(self) -> RAPDU:
...
         ticker_to_conf = {
+            "ETC": (ETC_CONF, ETC_CONF_DER_SIGNATURE, ETC_PACKED_DERIVATION_PATH),
             "ETH": (ETH_CONF, ETH_CONF_DER_SIGNATURE, ETH_PACKED_DERIVATION_PATH),
             "BTC": (BTC_CONF, BTC_CONF_DER_SIGNATURE, BTC_PACKED_DERIVATION_PATH),
             "LTC": (LTC_CONF, LTC_CONF_DER_SIGNATURE, LTC_PACKED_DERIVATION_PATH),
         }
-        assert ticker in ticker_to_conf
-        conf, signature, derivation_path = ticker_to_conf[ticker]
+        assert ticker.upper() in ticker_to_conf
+        conf, signature, derivation_path = ticker_to_conf[ticker.upper()]
```
Comment (fbeutin-ledger) [inline test/python/apps/exchange.py:196]: Why the `upper` ?
Eventually we may want to check the ticker against the values of  COIN_CONF[1:3] which contains the real ticker bytes sent to the app (full case here)

**Review by fbeutin-ledger — APPROVED** [review]
good to go once conversations are feedbacked

File: src/checks.h (hunk 2 above)
Comment (lpascal-ledger) [reply src/checks.h:2]: I don't like `#pragma once` as I already had issue with samely-named files (in different directories) and the pragma can't deal with it. It does not apply here, but every other header uses macros and I dislike inconsistency. :shrug:

File: CHANGELOG.md (hunk 1 above)
Comment (lpascal-ledger) [reply CHANGELOG.md:16]: Frankly, that'll take much longer that what I'm ready to offer to exchange rn.

File: test/python/apps/exchange.py (hunk 3 above)
Comment (lpascal-ledger) [reply test/python/apps/exchange.py:196]: LL APDUs keep using lower case versions of the tickers, and it seems to work, so I figured we don't care at that point.
But you're right I better check it thoroughly.

Comment (lpascal-ledger) [reply test/python/apps/exchange.py:196]: Confirmed, that's okay: transaction are normalized [here](https://github.com/LedgerHQ/app-exchange/blob/develop/src/process_transaction.c#L65), so ticker could be lower or upper case.

**Pull request merged**